gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/Topic.java

```diff
@@ -73,70 +82,173 @@ public Builder toBuilder() {
   }
 
   @Override
-  public int hashCode() {
+  public final int hashCode() {
     return Objects.hash(options, super.hashCode());
   }
 
   @Override
-  public boolean equals(Object obj) {
-    if (this == obj) {
+  public final boolean equals(Object obj) {
+    if (obj == this) {
       return true;
     }
-    if (obj == null || getClass() != obj.getClass()) {
+    if (obj == null || !obj.getClass().equals(Topic.class)) {
```

Comment:
I changed only for consistency with other classes across `gcloud-java` (see [Zone](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-dns/src/main/java/com/google/cloud/dns/Zone.java#L203) or [Blob](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-storage/src/main/java/com/google/cloud/storage/Blob.java#L532) for example).